metals/src/main/scala/scala/meta/internal/metals/OnTypeFormattingProvider.scala

```diff
@@ -8,17 +8,35 @@ import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.internal.mtags.Semanticdbs
 import scala.meta.tokens.Token
 import scala.meta.tokens.Token.Constant
+import scala.meta.tokens.Tokens
 
 /*in order to use onTypeFormatting in vscode,
 you'll have to set editor.formatOnType = true in settings*/
-
 final class OnTypeFormattingProvider(
     semanticdbs: Semanticdbs,
     buffer: Buffers
 )(implicit ec: ExecutionContext) {
 
-  val tripleQuote = """\u0022\u0022\u0022"""
-  val space = " "
+  private val tripleQuote = """\u0022\u0022\u0022"""
+  private val space = " "
+  private val stripMargin = "stripMargin"
+
+  private def isFinishedByStripMargin(
+      stringTokenIndex: Int,
+      tokens: Tokens
+  ): Boolean = {
+    var methodIndex = stringTokenIndex + 1
+
+    while (tokens(methodIndex).isWhiteSpaceOrComment ||
```

Comment:
Does this handle interpolated strings?

```
s"""
  |Hello $world
""".stripMargin
```

Reply:
Yes, however I don't think there is a test case for it. I will add that.

Reply:
I just tried the latest snapshot and the `|` addition didn't work for me when using an interpolator. Does this PR fixes it?

Reply:
Och, I think there might be an issue actually coming to think of that 🙁 Will add test cases and fix it here.

Reply:
Ok, now I fixed it. I just assumed we tested it.